gradle-plugin/generate_build_properties.gradle

```diff
@@ -32,4 +33,13 @@ def generateBuildProperties = project.tasks.register('generateBuildProperties')
 
 compileKotlin {
   dependsOn generateBuildProperties
-}
\ No newline at end of file
+}
+
+// Add the task dependencies to enforce correct ordering when depending out outputs.
+tasks.whenTaskAdded {
+  // For some reason when using `tasks.named('sourcesJar')` Gradle complains that the task cannot
+  // be found. Probably a lifecycle issue with afterEvaluate...
+  if (it.name == 'sourcesJar' || it.name == 'dokkaHtml') {
+    it.dependsOn(generateBuildProperties)
+  }
+}
```

Comment:
Gradle complained, so I fixed it.
```
> Task :dokkaHtml
Execution optimizations have been disabled for task ':dokkaHtml' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/Users/ralf/Development/projects/square/anvil/gradle-plugin/build/generated/sources/build-properties/kotlin/main'. Reason: Task ':dokkaHtml' uses this output of task ':generateBuildProperties' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```